src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/sentiment/WordListSentimentDetector.scala

```diff
@@ -0,0 +1,77 @@
+package com.microsoft.partnercatalyst.fortis.spark.transforms.sentiment
+
+import java.io.{File, IOError, IOException}
+import java.util.concurrent.ConcurrentHashMap
+
+import com.microsoft.partnercatalyst.fortis.spark.logging.Loggable
+import com.microsoft.partnercatalyst.fortis.spark.transforms.ZipModelsProvider
+import com.microsoft.partnercatalyst.fortis.spark.transforms.nlp.Tokenizer.tokenize
+import com.microsoft.partnercatalyst.fortis.spark.transforms.sentiment.SentimentDetector.{NEGATIVE, NEUTRAL, POSITIVE}
+
+import scala.io.Source
+
+@SerialVersionUID(100L)
+class WordListSentimentDetector(
+  modelsSource: Option[String] = None
+) extends Serializable with Loggable {
+
+  @volatile private lazy val wordsCache = new ConcurrentHashMap[String, Set[String]]
+  @volatile private lazy val modelsProvider = createModelsProvider()
+
+  def detectSentiment(text: String, language: String): Option[Double] = {
+    try {
+      val resourcesDirectory = modelsProvider.ensureModelsAreDownloaded(language)
+      val words = tokenize(text.toLowerCase)
+      val numPositiveWords = countPositiveWords(language, words, resourcesDirectory)
+      val numNegativeWords = countNegativeWords(language, words, resourcesDirectory)
+      computeSentimentScore(numPositiveWords, numNegativeWords)
+    } catch {
+      case ex @ (_ : IOException | _ : IOError) =>
+        logError(s"Unable to extract sentiment for language $language", ex)
+        None
+    }
+  }
+
+  private def computeSentimentScore(numPositiveWords: Int, numNegativeWords: Int) = {
+    if (numPositiveWords > numNegativeWords) {
```

Comment:
I'm not a fan of sprinkling Scala constructs left right and center which make the code harder to understand for folks coming from more mainstream languages.